### PR TITLE
[rootcling]Simplify dictionary stream splitting logic

### PR DESCRIPTION
**Pull request by reikdas**
Fixes failing clang-tidy travis instance

@vgvassilev

### CODE REVIEW

File: core/dictgen/src/rootcling_impl.cxx [hunk 1]
```diff
@@ -4519,38 +4519,29 @@ int RootClingMain(int argc,
...
          }
-      } else {
-         dictStreamPtr = &std::cout;
       }
 
       // Now generate a second stream for the split dictionary if it is necessary
       if (gOptSplit) {
-         splitDictStreamPtr = CreateStreamPtrForSplitDict(gOptDictionaryFileName.getValue(), tmpCatalog);
-         splitDeleter.reset(splitDictStreamPtr);
+         splitDictStream = CreateStreamPtrForSplitDict(gOptDictionaryFileName.getValue(), tmpCatalog);
```
Comment (pcanal) [inline core/dictgen/src/rootcling_impl.cxx:4540]: In the new version, where/when is the ofstream allocated by CreateStreamPtrForSplitDict deleted?

File: core/dictgen/src/rootcling_impl.cxx [hunk 2]
```diff
@@ -4519,38 +4519,31 @@ int RootClingMain(int argc,
    // Check if code goes to stdout or rootcling file
    std::ofstream fileout;
    string main_dictname(gOptDictionaryFileName.getValue());
-   std::ostream *dictStreamPtr = NULL;
-   std::ostream *dictStream = nullptr;
+   std::ostream *dictStream = &std::cout;
```
Comment (vgvassilev) [inline core/dictgen/src/rootcling_impl.cxx:4522]: We can make that a `std::ostream&`

File: core/dictgen/src/rootcling_impl.cxx (hunk 1 above)
Comment (reikdas) [reply core/dictgen/src/rootcling_impl.cxx:4541]: Oops, my bad. Thanks for catching it! Should be okay now.

File: core/dictgen/src/rootcling_impl.cxx (hunk 2 above)
Comment (pcanal) [reply core/dictgen/src/rootcling_impl.cxx:4522]: Are you sure?  Isn't it sometime swapped for a file/ofstream?

Comment (reikdas) [reply core/dictgen/src/rootcling_impl.cxx:4522]: Like @pcanal said, dictstream is assigned to fileout here - https://github.com/root-project/root/pull/5345/files#diff-64a4c9a2b092d1b125517e200783ca9fR4531 so I am not sure if that is possible..

Comment (vgvassilev) [reply core/dictgen/src/rootcling_impl.cxx:4522]: Indeed the assignment op is deleted. We could use the ternary op:
`std::ostream& ref = (some_condition) ? fileout : std::cout;`

**Comment by vgvassilev** [comment]
Note that Jenkins has not yet run.

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by vgvassilev** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Pull request merged**